sds_data_manager/lambda_code/SDSCode/opensearch_utils/snapshot.py

```diff
@@ -0,0 +1,101 @@
+import logging
+import string
+from datetime import datetime
+
+import boto3
+import requests
+from requests_aws4auth import AWS4Auth
+
+
+def get_auth(region):
+    # Get AWS service and credentials for snapshot
+    service = "es"
+    credentials = boto3.Session().get_credentials()
+    awsauth = AWS4Auth(
+        credentials.access_key,
+        credentials.secret_key,
+        region,
+        service,
+        session_token=credentials.token,
+    )
+
+    return awsauth
+
+
+def register_repo(payload: dict, url: string, awsauth):
+    """Register the snapshot repo
```

Comment:
Can you add some comments about repo? Repo is used for so many things that it's confusing sometimes. Is it ECR repo?